library/alloc/src/string.rs

```diff
@@ -1849,6 +1849,34 @@ impl String {
         let slice = self.vec.into_boxed_slice();
         unsafe { from_boxed_utf8_unchecked(slice) }
     }
+
+    /// Consumes and leaks the `String`, returning a mutable reference to the contents,
+    /// `&'a mut str`.
+    ///
+    /// This function is mainly useful for data that lives for the remainder of
+    /// the program's life. Dropping the returned reference will cause a memory
+    /// leak.
+    ///
+    /// # Examples
+    ///
+    /// Simple usage:
+    ///
+    /// ```
+    /// #![feature(string_leak)]
+    ///
+    /// pub fn main() {
+    ///     let x = String::from("bucket");
+    ///     let static_ref: &'static mut str = x.leak();
+    ///     assert_eq!(static_ref, "bucket");
+    /// }
```

Comment:
There's no need for an `fn main()` in tests:
```suggestion
    /// let x = String::from("bucket");
    /// let static_ref: &'static mut str = x.leak();
    /// assert_eq!(static_ref, "bucket");
```